common/src/ethereum/walletconnect.rs

```diff
@@ -0,0 +1,104 @@
+use std::error::Error;
+use walletconnect::client::{CallError, SessionError};
+use walletconnect::{qr, Client, Metadata, Transaction};
+
+use ethers::types::transaction::eip2718::TypedTransaction;
+use ethers::types::{Address, NameOrAddress, Signature, U256};
+
+#[derive(Debug)]
+pub struct WalletConnect {
+    client: Client,
+    chain_id: u64,
+    address: Address,
+}
+
+impl WalletConnect {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
+        let client = Client::new(
+            "rad-ens",
+            Metadata {
+                description: "Change ENS records for your Radicle profile".into(),
+                url: "https://github.com/radicle-dev".parse()?,
+                icons: vec!["https://avatars0.githubusercontent.com/u/48290027".parse()?],
+                name: "Radicle CLI".into(),
+            },
+        )?;
+
+        Ok(WalletConnect {
+            client,
+            chain_id: 0,
+            address: Address::zero(),
+        })
+    }
+
+    pub async fn show_wait_connection(mut self) -> Result<Self, SessionError> {
```

Comment:
```suggestion
    pub async fn show_qr(mut self) -> Result<Self, SessionError> {
```
Clearer to me

Reply:
https://github.com/radicle-dev/radicle-cli/commit/0028efb90cc8297feef23f19dc3f55cd071afb6e#diff-363b2cf3885221c5b26bfa5862fc3e2363454ef177a777c5e11759468b8edc5cR34